recipes/prometheus-cpp/all/conanfile.py

```diff
@@ -0,0 +1,120 @@
+from conans import ConanFile, CMake, tools
+import os
+
+
+class PrometheusCppConan(ConanFile):
+    name = "prometheus-cpp"
+    description = "Prometheus Client Library for Modern C++"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/jupp0r/prometheus-cpp"
+    license = "MIT"
+    topics = ("conan", "metrics", "prometheus", "networking")
+    exports_sources = ["CMakeLists.txt"]
+    generators = "cmake", "cmake_find_package", "cmake_find_package_multi"
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_pull": [True, False],
+        "with_push": [True, False],
+        "with_compression": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_pull": True,
+        "with_push": True,
+        "with_compression": True,
+    }
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        tools.check_min_cppstd(self, 11)
+
+    def requirements(self):
+        if self.options.with_pull:
+            self.requires("civetweb/1.13")
+        if self.options.with_push:
+            self.requires("libcurl/7.74.0")
+        if self.options.with_compression:
+            self.requires("zlib/1.2.11")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["USE_THIRDPARTY_LIBRARIES"] = False
+        self._cmake.definitions["ENABLE_TESTING"] = False
+        self._cmake.definitions["OVERRIDE_CXX_STANDARD_FLAGS"] = False
+
+        self._cmake.definitions["BUILD_SHARED_LIBS"] = self.options.shared
+        self._cmake.definitions["ENABLE_PULL"] = self.options.with_pull
+        self._cmake.definitions["ENABLE_PUSH"] = self.options.with_push
+        self._cmake.definitions["ENABLE_COMPRESSION"] = self.options.with_compression
```

Comment:
What if libcurl does not have compression is that a problem? https://github.com/conan-io/conan-center-index/blob/8c69557071cf79e94a6807c7b44a101c843da6ea/recipes/libcurl/all/conanfile.py#L152

Reply:
This does not matter since lib curl is push